include/solvers/time_solver.h

```diff
@@ -261,12 +280,33 @@ class TimeSolver : public ReferenceCountedObject<TimeSolver>
   { _is_adjoint = _is_adjoint_value; }
 
   /**
-   * Function to return 'last_deltat()', returns system.deltat if
-   * fixed timestep solver is used, completed_deltat if the adaptive
+   * Returns system.deltat if fixed timestep solver is used,
+   * the complete timestep size (sum of all substeps) if the adaptive
    * time solver is used.
    */
-  virtual Real last_complete_deltat();
+  virtual Real last_completed_timestep_size();
+
+  /**
+   * Returns the system.deltat timestep used to get to the current time.
+   * For a uniform timestepper, it will simply return the last used system.deltat, for an
+   * adaptive time solver, it will return the last used substep size.
+   */
+  virtual Real last_completed_deltat();
```

Comment:
And this smells like we're doing something wrong.  This is supposed to be the TimeSolver class.  Users at this level should be protected from digging into N details of N different subclasses except where that's absolutely unavoidable.

This description doesn't seem to match what you've got going on in the new example, either.  The time advanced by a solve is supposed to be the complete deltat size, not the size of the last substep.

Reply:
Purged this function and replaced its use with last_completed_timestep_size in adjoints example 5.